### PR TITLE
feat(journey-maps): switch between 2D and 3D layers

### PR DESCRIPTION
**Pull request by swiss-chris**
depdnging on the levelSwitch uiOptions

Refs: ROKAS-1078

### CODE REVIEW

File: src/journey-maps/angular/services/map/two-d-three-d-service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  /**
+   * If the level-switch feature is enabled by the client, then show only 3d layers (-lvl)
+   * If the level-switch feature is disabled by the client, then show only 2d layers (-2d)
+   */
+  public show2Dor3D(map: MaplibreMap, isLevelSwitchEnabled: boolean) {
+    this._setVisibility(map, '-2d', isLevelSwitchEnabled ? 'none' : 'visible');
+    this._setVisibility(map, '-lvl', isLevelSwitchEnabled ? 'visible' : 'none');
+  }
+
+  private _setVisibility(map: MaplibreMap, searchString1: string, newVisibility1: string) {
```
Comment (sebisteiner) [inline src/journey-maps/angular/services/map/two-d-three-d-service.ts:17]:
```suggestion
  private _setVisibility(map: MaplibreMap, layerIdSuffix: string, visibility: 'visible' | 'none') {
```

**Review by sebisteiner — DISMISSED** [review]

**Comment by besartmemeti** [comment]
LGTM

File: src/journey-maps/angular/services/map/two-d-three-d-service.ts (hunk 1 above)
Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts]: Done ✅

Comment (sebisteiner) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts]: Don't you like `visibility: 'visible' | 'none'` 🥲

**Review by sebisteiner — APPROVED** [review]
LGTM 👍

File: src/journey-maps/angular/services/map/two-d-three-d-service.ts (hunk 1 above)
Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts]: Sorry, das hatte ich nicht gesehen. Hab's gepushed.

File: src/journey-maps/angular/services/map/two-d-three-d-service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { Map as MaplibreMap } from 'maplibre-gl';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TwoDThreeDService {
```
Comment (kyubisation) [inline src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: Out of curiosity; Does this functionality really need a separate class for it? The overhead for bundling, runtime and just developer mindspace seems barely worth it?

Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: As it currently stands, this service is only called from one single method, so yes, we could inline this code to the same class as the method currently calling it. Let me know what you prefer and I can refactor it in a few minutes.

Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: In general, what kind of criteria do/would you use to decide whether or not to create a separate service ?

Comment (kyubisation) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: As usual, it's not exactly obvious what the right decision is.
I think in this specific case, I would prefer to inline this functionality.
From my perspective, there is no merit in separating this code into its own class.
But as mentioned, YMMV.

Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: I'll push the change in a minute after bazel lets me test it :)

Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts:7]: > I'll push the change in a minute after bazel lets me test it :)

https://xkcd.com/303/

Comment (swiss-chris) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts]: > I'll push the change in a minute after bazel lets me test it :)

Done ✅

**Review by kyubisation — APPROVED** [review]
LGTM 👍

File: src/journey-maps/angular/services/map/two-d-three-d-service.ts (hunk 2 above)
Comment (kyubisation) [reply src/journey-maps/angular/services/map/two-d-three-d-service.ts]: > In general, what kind of criteria do/would you use to decide whether or not to create a separate service ?

I'm afraid I don't have a clear answer.
There is the difference between library and product development. In product development you are far more free in structuring your app, but in library development you have to pay attention to a lot more of attention (e.g. performance on component level and every separate class is a slight overhead).
So, generally, do what feels right. But I might ask questions if I have doubts. 😃

**Pull request merged**